src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -415,9 +415,24 @@ public static TransportMessage ExtractTransportMessage(Message receivedMessage,
             }
             // HINT: Message Id is the only required header
             transportMessage.Headers.TryAdd(Headers.MessageId, messageIdOverride);
+            AddCustomNativeHeadersToNServiceBusHeaders(receivedMessage, transportMessage);
+
             return transportMessage;
         }
 
+        static void AddCustomNativeHeadersToNServiceBusHeaders(Message receivedMessage, TransportMessage transportMessage)
+        {
+            foreach (var receivedMessageMessageAttribute in receivedMessage.MessageAttributes)
+            {
+                // TODO is there a better way to achieve the same result? This seems fragile
```

Comment:
You can probably filter the ones out you don't want with a linq query, which will improve readability too, but possibly it's slower (that would be Daniel's department ;-))

Reply:
@danielmarbach @lailabougria we need to discuss the `if` in this `foreach`. It's super fragile.

Reply:
Like I mentioned in slack, I think we can override all matching ones with the native. That would be consistent with the description we have in the doc following the principle of native headers win